voxblox/include/voxblox/core/layer_inl.h

```diff
@@ -39,6 +39,36 @@ Layer<VoxelType>::Layer(const LayerProto& proto)
   voxels_per_side_inv_ = 1.0f / static_cast<FloatingPoint>(voxels_per_side_);
 }
 
+template <typename VoxelType>
+void Layer<VoxelType>::forEachVoxelInLayer(
+    const VoxelAction& voxel_action) const {
+  voxblox::BlockIndexList all_blocks;
+  getAllAllocatedBlocks(&all_blocks);
+  for (const voxblox::BlockIndex& block_index : all_blocks) {
+    const BlockType& block = getBlockByIndex(block_index);
+    for (size_t linear_voxel_index = 0u;
+         linear_voxel_index < block.num_voxels(); ++linear_voxel_index) {
+      const VoxelType& voxel = block.getVoxelByLinearIndex(linear_voxel_index);
+      voxel_action(block_index, linear_voxel_index, block, voxel);
+    }
+  }
+}
+
+template <typename VoxelType>
+void Layer<VoxelType>::forEachVoxelInLayer(
+    const MutableVoxelAction& voxel_action) {
+  voxblox::BlockIndexList all_blocks;
+  getAllAllocatedBlocks(&all_blocks);
+  for (const voxblox::BlockIndex& block_index : all_blocks) {
+    BlockType& block = *CHECK_NOTNULL(getBlockPtrByIndex(block_index));
```

Comment:
Could you use
```
BlockType& block = getBlockByIndex(block_index);
```
as above (just without `const`)?

...`getBlockByIndex` will end in FATAL if the block does not exist, covering your `CHECK_NOTNULL`.